integration-tests/contract-permits-set.spec.ts

```diff
@@ -0,0 +1,141 @@
+        'y4BX7qS1UE',
+        [
+          'skate',
+          'damp',
+          'faculty',
+          'morning',
+          'bring',
+          'ridge',
+          'traffic',
+          'initial',
+          'piece',
+          'annual',
+          'give',
+          'say',
+          'wrestle',
+          'rare',
+          'ability',
+        ].join(' '),
+        '7d4c8c3796fdbf4869edb5703758f0e5831f5081'
+      );
+        done()
+    })
+
+    test('Create permit hash for an fa2 contract', async (done) => {
+
+      const op = await Tezos.contract.originate({
+        code: permit_admin_42_set,
+        storage:
+        {
+        0: new MichelsonMap(),
+        1: 300,
+        2: bob_address,
+        3: bob_address
+        },
+        });
+        await op.confirmation();
+        expect(op.hash).toBeDefined();
+        expect(op.includedInBlock).toBeLessThan(Number.POSITIVE_INFINITY);
+        const contract = await op.contract();
+        expect(op.status).toEqual('applied')
+        const permit_address = (await op.contract()).address
+
+        // Get the contract
+        const permit_contract = await Tezos.contract.at(permit_address);
+       
+        // Check whether bob is actually the admin
+        const storage : any = await permit_contract.storage();
+
+        // Get the signer's public key and a dummy signature to trigger the error
+        const signer_key = await Tezos.signer.publicKey().catch(e => console.error(e));
+        const dummy_sig = "edsigu5scrvoY2AB7cnHzUd7x7ZvXEMYkArKeehN5ZXNkmfUSkyApHcW5vPcjbuTrnHUMt8mJkWmo8WScNgKL3vu9akFLAXvHxm";
+       
+        // Get the Blake2B hash of the packed parameter
+        const wrapped_param : any = permit_contract.methods['wrapped'](42).toTransferParams().parameter?.value;
+        const wrapped_param_type = contract.entrypoints.entrypoints['wrapped'];       
+        const raw_packed = await Tezos.rpc.packData({
+          data: wrapped_param,
+          type: wrapped_param_type,
+        }).catch(e => console.error('error:', e));
+        var packed_param;
+        if (raw_packed) {
+          packed_param = raw_packed.packed
+        } else {
+          throw `packing ${wrapped_param} failed`
+        };
+          
+        const param_hash =  buf2hex(blake.blake2b(hex2buf(packed_param), null, 32));
+
+        const expected_param_hash = "0f0db0ce6f057a8835adb6a2c617fd8a136b8028fac90aab7b4766def688ea0c";
+        if(param_hash === expected_param_hash) {
+        } else {
+          throw `unexpected param_hash: {param_hash},\n
+          while {expected_param_hash} was expected`;
+        }
+        
+        // Preapply a transfer with the dummy_sig to extract the bytes_to_sign
+        const transfer_params = permit_contract.methods.permit(signer_key, dummy_sig, param_hash).toTransferParams();
+        const bytes_to_sign = await Tezos.estimate.transfer(transfer_params).catch((e) => errors_to_missigned_bytes(e.errors));
```

Comment:
So this line fails and the bytes to sign are retrieved from the error message. There is a similarity with the lambda view here :+1:

Reply:
This was from the example from Michael J. Klein